contracts/verifiers/PunksVerifier.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.11;
+
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+import "../interfaces/ISignatureVerifier.sol";
+import "../external/interfaces/IPunks.sol";
+
+import { IV_InvalidTokenId } from "../errors/Lending.sol";
+
+/**
+ * @title PunksVerifier
+ * @author Non-Fungible Technologies, Inc.
+ *
+ * See ItemsVerifier for a more thorough description of the Verifier
+ * pattern used in Arcade.xyz's lending protocol. This contract
+ * verifies predicates that check ownership of a certain CryptoPunk
+ * by an asset vault.
+ */
+contract PunksVerifier is ISignatureVerifier {
+    using SafeCast for int256;
+
+    // ============================================ STATE ==============================================
+
+    // =============== Contract References ===============
+
+    IPunks public immutable punks;
+
+    // ========================================== CONSTRUCTOR ===========================================
+
+    constructor(address _punks) {
+        punks = IPunks(_punks);
+    }
+
+    // ==================================== COLLATERAL VERIFICATION =====================================
+
+    /**
+     * @notice Verify that the items specified by the packed uint256 array are held by the vault.
```

Comment:
is the array a packed `uint256` or` int256` ?

Reply:
Fixed line 44